module/zfs/dsl_scan.c

```diff
@@ -701,6 +704,11 @@ dsl_scan_recurse(dsl_scan_t *scn, dsl_dataset_t *ds, dmu_objset_type_t ostype,
 		int epb = BP_GET_LSIZE(bp) >> DNODE_SHIFT;
 		arc_buf_t *buf;
 
+		if (BP_IS_ENCRYPTED(bp)) {
+			ASSERT3U(BP_GET_COMPRESS(bp), ==, ZIO_COMPRESS_OFF);
+			zio_flags |= ZIO_FLAG_RAW;
+		}
```

Comment:
Ah, right, because this is where we actually look at the contents.  I was confused though this was the scrub i/o.  Should this be `ZIO_FLAG_RAW_ENCRYPT` then?